segmenters/nlp/SpacySentencizer/__init__.py

```diff
@@ -0,0 +1,97 @@
+from typing import Dict, List
+
+from jina.executors.segmenters import BaseSegmenter
+
+
+class SpacySentencizer(BaseSegmenter):
+    """
+    :class:`SpacySentencizer` split the text on the doc-level
+    into sentences on the chunk-level with spaCy as the backend.
+    The splitting technique is divided into two different approaches:
+    1) Default Segmenter: Utilizes dependency parsing to determine the rule for
+        splitting the text. For more info please refer to https://spacy.io/api/sentencizer.
+    2) Machine Learning-based Segmenter: Utilizes SentenceRecognizer model trained
+        on a dedicated data created for sentence segmentation. For more info please
+        refer to https://spacy.io/api/sentencerecognizer.
+
+    :param lang: pre-trained spaCy language pipeline
+    :param use_default_segmenter: if True will use dependency based sentence segmentation,
+        otherwise ml-based implementation will be chosen,
+        by default False.
+    :param args:  Additional positional arguments
+    :param kwargs: Additional keyword arguments
+    """
+
+    SPACY_COMPONENTS = [
+        'tagger',
+        'parser',
+        'ner',
+        'senter',
+        'entity_linker',
+        'textcat',
+        'entity_ruler',
+        'merge_noun_chunks',
+        'merge_entities',
+        'merge_subtokens',
+        'lemmatizer',
+        'attribute_ruler',
+    ]
+
+    def __init__(self, lang: str, use_default_segmenter: bool = False, *args, **kwargs):
```

Comment:
we need to provide a proper default value for `lang`

Reply:
for us to be able to test the default

Reply:
I tried to do that by using multilingual model, but the problem is we need to download the model separately by adding the following line to requirements.txt
`https://github.com/explosion/spacy-models/releases/download/<LANG>-3.0.0/<LANG>-3.0.0.tar.gz`
As you can see in the beginning of this PR, it will introduce error as the line can't be parsed.

There is another way, not sure if this is considered as hack or not, by calling `spacy.cli.download` in the constructor. However, that function is actually just a wrapper for subprocess and if the value of `lang` is not available in spaCy it will return system exit error instead of just the usual Exception error.

For now I remove the default value as it seems weird to me if we can't install the model outside dockerfile and enforce default value for `lang`.

Let me know what'd you think about it.